metricbeat/scripts/msetlists/main.go

```diff
@@ -0,0 +1,49 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	_ "github.com/elastic/beats/metricbeat/include"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+func main() {
+
+	// List all registered modules and metricsets.
+	var defaultMap = make(map[string][]string)
+	for _, mod := range mb.Registry.Modules() {
+		metricSets, err := mb.Registry.DefaultMetricSets(mod)
+		if err != nil && !strings.Contains(err.Error(), "no default metricset for") {
+			continue
+		}
+		defaultMap[mod] = metricSets
+	}
+
+	raw, err := json.MarshalIndent(defaultMap, "", "  ")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error Marshalling json: %s\n", err)
+		os.Exit(1)
+	}
```

Comment:
We could move this common code to a common function so we don't forget to update one if we update the other.

Reply:
Yah, I would ideally like to make most of this a common library, but I honestly wasn't sure where to put it.